Tests/BSONTests/BSONCorpusTests.swift

```diff
@@ -99,124 +98,114 @@ final class BSONCorpusTests: MongoSwiftTestCase {
-                // CDRIVER-2223, legacy extended JSON $date syntax uses numbers
-                "Bad $date (number, not string or hash)",
-                // TODO: SWIFT-330: unskip these
-                "Invalid BSON type low range",
-                "Invalid BSON type high range"
-            ]
+        let INCLUDED_CORPUS_TESTS = [
+            "Int32 type",
+            "Int64 type"
         ]
 
-        let shouldSkip = { testFileDesc, testDesc in
-            SKIPPED_CORPUS_TESTS[testFileDesc]?.contains { $0 == testDesc } == true
+        let shouldRun: (String, String) -> Bool = { testFileDesc, testDesc in
+            INCLUDED_CORPUS_TESTS.contains(testFileDesc)
         }
 
         for (_, testFile) in try retrieveSpecTestFiles(specName: "bson-corpus", asType: BSONCorpusTestFile.self) {
             if let validityTests = testFile.valid {
-                for test in validityTests {
-                    guard !shouldSkip(testFile.description, test.description) else {
-                        continue
-                    }
-
+                for test in validityTests where shouldRun(testFile.description, test.description) {
                     guard let cBData = Data(hexString: test.canonicalBSON) else {
                         XCTFail("Unable to interpret canonical_bson as Data")
                         return
                     }
 
-                    guard let cEJData = test.canonicalExtJSON.data(using: .utf8) else {
-                        XCTFail("Unable to interpret canonical_extjson as Data")
-                        return
-                    }
+                    // guard let cEJData = test.canonicalExtJSON.data(using: .utf8) else {
+                    //     XCTFail("Unable to interpret canonical_extjson as Data")
+                    //     return
+                    // }
 
-                    let lossy = test.lossy ?? false
+                    // let lossy = test.lossy ?? false
 
                     // for cB input:
                     // native_to_bson( bson_to_native(cB) ) = cB
-                    let docFromCB = try Document(fromBSON: cBData)
-                    expect(docFromCB.rawBSON).to(equal(cBData))
+                    let docFromCB = try BSONDocument(fromBSON: cBData)
+                    expect(docFromCB.toData()).to(equal(cBData))
 
                     // test round tripping through documents
                     // We create an array by reading every element out of the document (and therefore out of the
-                    // bson_t). We then create a new document and append each element of the array to it. Once that
-                    // is done, every element in the original document will have gone from bson_t -> Swift data type
-                    // -> bson_t. At the end, the new bson_t should be identical to the original one. If not, our bson_t
-                    // translation layer is lossy and/or buggy.
-                    let nativeFromDoc = docFromCB.toArray()
```

Comment:
I think we still want to be using `toArray()` and `init(fromArray:)` here. they are subtly different. 
`toArray()` recursively calls the same thing on sub-document, ensuring we fully decode through all layers of nested data. `Array(...)` will only do that at the top level, and nested documents will not get fully decoded into Swift types.

I think you are missing these at the moment because they are defined in the driver's `DocumentTests.swift`: https://github.com/mongodb/swift-bson/blob/master/Tests/DisabledTests/DocumentTests.swift#L32-L64

they are actually only used by the corpus runner - at some point we moved the corpus runner to its own file but forgot to bring the helpers over. so I think you could just move them over in this PR and then switch back to using them here.

Reply:
Ah good call, I figured this was API on the actual BSONDocument I hadn't done yet, I pulled in the helpers but I can't use them until I make document's mutable which has a ticket so marked this TODO